extensions/src/main/java/io/lionweb/serialization/extensions/NodeInfo.java

```diff
@@ -0,0 +1,57 @@
+package io.lionweb.serialization.extensions;
+
+import java.util.Objects;
+import org.jetbrains.annotations.Nullable;
+
+public final class NodeInfo {
```

Comment:
Maybe add a bit of docs explaining the difference between `NodeInfo`, `SerializedNode`, etc?

Reply:
I added a comment for this class